tools/integration-test/src/tests/forward/forward_transfer.rs

```diff
@@ -0,0 +1,418 @@
+//! This test tests three different cases:
+//!
+//! - The `IbcForwardTransferTest` tests the case a packet is successfully
+//!   forwarded.
+//!
+//! - The `InvalidChannelIbcForwardTransferTest` tests the case where the
+//!   sender is refunded if the forward channel specified is invalid.
+//!
+//! - The `InvalidAddressIbcForwardTransferTest` tests the case when the destination
+//!   address is invalid. This results in two scenarios depending on the Gaia
+//!   version.
+//!   For Gaia 6 the intermediate address will receive the refunded tokens.
+//!   For Gaia 8 the sender will receive the refunded tokens.
```

Comment:
I understand that this is the behavior exhibited by Gaia versions 6 and 8, but is this behavior intended for the respective Gaia versions, or are these bugs in the packet forwarding behavior for these versions? 

I'm mostly wondering if this test will need to be modified in the future in the case that this behavior is made uniform across Gaia versions.

Reply:
I am not sure the behaviour of Gaia 6 is the one which was expected since the atomic forwarding isn't implemented. Gaia 6 uses the v1.0.1 forward middleware module, and atomic forwarding was introduced in v3. But the behaviour of Gaia 8 and I suppose all the next ones shouldn't change.
I will added more information in the comments regarding this.